bricks/advanced/src/data-providers/snapshot.ts

```diff
@@ -0,0 +1,83 @@
+// istanbul ignore file: nothing logical except calling html2canvas.
+import { http } from "@next-core/http";
+import html2canvas from "html2canvas";
+
+export type UploadStatus = "uploading" | "done" | "error";
+
+export function capture(id?: string): Promise<HTMLCanvasElement> {
+  const target = document.querySelector(`#${id}`) || document.body;
+
+  return new Promise(function (resolve, reject) {
+    html2canvas(document.body, {
+      logging: false,
+      scale: window.devicePixelRatio < 3 ? window.devicePixelRatio : 2,
+      width: target.clientWidth || window.innerWidth,
+      height: target.clientHeight || window.innerHeight,
```

Comment:
这个处理，如果 target 找到了，但是其高宽为 0，会导致截图高宽变成了整个窗口，不太符合预期，所以这样好点：
```suggestion
      width: target ? target.clientWidth : window.innerWidth,
      height: target ? target.clientHeight : window.innerHeight,
```